src/goto-programs/goto_program.cpp

```diff
@@ -634,3 +634,45 @@ void goto_programt::compute_incoming_edges()
   }
 }
 
+bool goto_programt::instructiont::equals(const instructiont &other) const
+{
+  // clang-format off
+  return
+    type == other.type &&
+    code == other.code &&
+    guard == other.guard &&
+    targets.size() == other.targets.size() &&
+    labels == other.labels;
+  // clang-format on
```

Comment:
This might be confusing as the target doesn't get compared?

Reply:
In my opinion that's not an aspect of equality that can be resolved on a single instruction. That's contextual information that only makes sense when looking at the program, I'd say.

Reply:
On a second thought: maybe the name of the function should be changed to convey this? And suggestions?

Reply:
See the second commit in this PR.